pvlib/pvsystem.py

```diff
@@ -1839,45 +1856,87 @@ def sapm_spectral_loss(airmass_absolute, module):
 
 
 def sapm_effective_irradiance(poa_direct, poa_diffuse, airmass_absolute, aoi,
-                              module, reference_irradiance=1000):
-    """
+                              module, reference_irradiance=1.):
```

Comment:
If I understand correctly, the function returns the same result as it used to if the function is called without `reference_irradiance`. But if the user did set `reference_irradiance`, then this code will always return a different result. I don't know if that's better or worse than a clean API break.

Reply:
You understand correctly, if I understand you correctly. 

Old behavior: inputs in W/m2, output in suns, `reference_irradiance=1000` W/m2. 

Interim behavior: input in W/m2, output in suns or W/m2, `reference_irradiance=1.` is used to signal the suns units on the output (which is a change I find awkward). An alternative is to change the default output unit to W/m2, `reference_irradiance=1000.` by default, and tell users to set `reference_irradiance=1.0` to get output in suns. That would change the default behavior.

Target Behavior after deprecation period: input and output in W/m2, and no reference irradiance needed, so that the output is ready as input for any of the DC power functions.

Reply:
I can't say I like any of the options, so I'm ok with just removing the kwarg and changing the API. In that case, you could make an argument that we shouldn't bother with the deprecation cycle for `sapm`.

Reply:
Neither did I. I changed back to W/m2 in/out in both functions, removed `reference_irradiance` from the effective irradiance function, and also removed `irrad_ref` and `temp_ref` as kwargs for `sapm`. We can expose those kwargs as `reference_irradiance` and `reference_temperature` when #825 renames throughout.